### PR TITLE
Text only edits to error message catalog (doc review)

### PR DESCRIPTION
**Pull request by MiriamScharnke**
First batch of error message review, including those edits that required no further code changes.

### CODE REVIEW

File: +llms/+utils/errorMessageCatalog.m [hunk 1]
```diff
@@ -41,22 +41,22 @@
...
-catalog("llms:mustBeValidIndex") = "Index exceeds the number of array elements. Index must be less than or equal to ({1}).";
-catalog("llms:stopSequencesMustHaveMax4Elements") = "Number of elements must not be larger than 4.";
+catalog("llms:mustBeValidIndex") = "Index exceeds the number of array elements. Index must be less than or equal to {1}.";
+catalog("llms:stopSequencesMustHaveMax4Elements") = "Number of stop sequences must be less than or equal to 4.";
 catalog("llms:endpointMustBeSpecified") = "Unable to find endpoint. Either set environment variable AZURE_OPENAI_ENDPOINT or specify name-value argument ""Endpoint"".";
 catalog("llms:deploymentMustBeSpecified") = "Unable to find deployment name. Either set environment variable AZURE_OPENAI_DEPLOYMENT or specify name-value argument ""Deployment"".";
 catalog("llms:keyMustBeSpecified") = "Unable to find API key. Either set environment variable {1} or specify name-value argument ""APIKey"".";
-catalog("llms:mustHaveMessages") = "Value must contain at least one message in Messages.";
+catalog("llms:mustHaveMessages") = "Message must not be empty.";
```
Comment (ccreutzi) [inline +llms/+utils/errorMessageCatalog.m:49]: An empty message in the list `Messages` would probably even be accepted. If you could get it in there.

```suggestion
catalog("llms:mustHaveMessages") = "Message history must not be empty.";
```

File: +llms/+utils/errorMessageCatalog.m [hunk 2]
```diff
@@ -41,22 +41,22 @@
...
-catalog("llms:invalidOptionForModel") = "{1} is not supported for ModelName '{2}'";
-catalog("llms:invalidContentTypeForModel") = "{1} is not supported for ModelName '{2}'";
-catalog("llms:functionNotAvailableForModel") = "This function is not supported for ModelName '{1}'";
-catalog("llms:promptLimitCharacter") = "Prompt must have a maximum length of {1} characters for ModelName '{2}'";
-catalog("llms:pngExpected") = "Argument must be a PNG image.";
+catalog("llms:mustBeMessagesOrTxt") = "Message must be nonempty string, character array, or messageHistory object.";
+catalog("llms:invalidOptionAndValueForModel") = "'{1}' with value '{2}' is not supported for model ""{3}"".";
+catalog("llms:invalidOptionForModel") = "Invalid argument name {1} for model ""{2}"".";
+catalog("llms:invalidContentTypeForModel") = "{1} is not supported for model ""{2}"".";
+catalog("llms:functionNotAvailableForModel") = "This function is not supported for model ""{1}"".";
```
Comment (ccreutzi) [inline +llms/+utils/errorMessageCatalog.m:55]: [Used in exactly one place](https://github.com/search?q=repo%3Amatlab-deep-learning%2Fllms-with-matlab%20llms%3AfunctionNotAvailableForModel&type=code), `openAIImages.edit`. I thought you wanted to make the text less generic for those cases?

```suggestion
catalog("llms:functionNotAvailableForModel") = "Image editing is not supported for model ""{1}"".";
```

File: +llms/+utils/errorMessageCatalog.m [hunk 3]
```diff
@@ -41,22 +41,22 @@
...
-catalog("llms:invalidOptionAndValueForModel") = "'{1}' with value '{2}' is not supported for ModelName '{3}'";
-catalog("llms:invalidOptionForModel") = "{1} is not supported for ModelName '{2}'";
-catalog("llms:invalidContentTypeForModel") = "{1} is not supported for ModelName '{2}'";
-catalog("llms:functionNotAvailableForModel") = "This function is not supported for ModelName '{1}'";
-catalog("llms:promptLimitCharacter") = "Prompt must have a maximum length of {1} characters for ModelName '{2}'";
-catalog("llms:pngExpected") = "Argument must be a PNG image.";
+catalog("llms:mustBeMessagesOrTxt") = "Message must be nonempty string, character array, or messageHistory object.";
+catalog("llms:invalidOptionAndValueForModel") = "'{1}' with value '{2}' is not supported for model ""{3}"".";
+catalog("llms:invalidOptionForModel") = "Invalid argument name {1} for model ""{2}"".";
+catalog("llms:invalidContentTypeForModel") = "{1} is not supported for model ""{2}"".";
```
Comment (ccreutzi) [inline +llms/+utils/errorMessageCatalog.m:54]: Only ever used with `"Image content"`. I guess that'll be changed when you also edit code?

**Review by ccreutzi — CHANGES_REQUESTED** [review]
Requesting change in `mustHaveMessages`.

File: +llms/+utils/errorMessageCatalog.m (hunk 3 above)
Comment (MiriamScharnke) [reply +llms/+utils/errorMessageCatalog.m:54]: Yes :) But that's useful to know, thank you!

**Review by ccreutzi — APPROVED** [review]

File: +llms/+utils/errorMessageCatalog.m [hunk 4]
```diff
@@ -41,22 +41,22 @@
...
+catalog("llms:mustHaveMessages") = "Message history must not be empty.";
 catalog("llms:mustSetFunctionsForCall") = "When no functions are defined, ToolChoice must not be specified.";
-catalog("llms:mustBeMessagesOrTxt") = "Messages must be text with one or more characters or a messageHistory object.";
-catalog("llms:invalidOptionAndValueForModel") = "'{1}' with value '{2}' is not supported for ModelName '{3}'";
-catalog("llms:invalidOptionForModel") = "{1} is not supported for ModelName '{2}'";
-catalog("llms:invalidContentTypeForModel") = "{1} is not supported for ModelName '{2}'";
-catalog("llms:functionNotAvailableForModel") = "This function is not supported for ModelName '{1}'";
-catalog("llms:promptLimitCharacter") = "Prompt must have a maximum length of {1} characters for ModelName '{2}'";
-catalog("llms:pngExpected") = "Argument must be a PNG image.";
+catalog("llms:mustBeMessagesOrTxt") = "Message must be nonempty string, character array, or messageHistory object.";
```
Comment (debymf) [inline +llms/+utils/errorMessageCatalog.m:51]:
```suggestion
catalog("llms:mustBeMessagesOrTxt") = "Message must be nonempty string, character array, cell array of character vectors, or messageHistory object.";
```

Comment (debymf) [reply +llms/+utils/errorMessageCatalog.m:51]: We also accept cell array of chars:
```
>> generate(mdl, {'hello'})

ans = 

    "Hello! How can I assist you today?"
```

**Review by debymf — APPROVED** [review]

File: +llms/+utils/errorMessageCatalog.m (hunk 4 above)
Comment (ccreutzi) [reply +llms/+utils/errorMessageCatalog.m]: We do not, however, accept arbitrary char arrays. (Plus, I have issue with calling chars “characters,” but I guess the doc standards just don't care about what I think. Not that they usually should.)

What's the proper doc term for a string that is not `""`? MATLAB does not regard `""` as “empty.”

**Pull request merged**